### DISCUSSION TITLE
Implementing Karate GQL Testing for Hasura

### DISCUSSION THREAD

**Discussion by shottah** [hasura/graphql-engine#9896] [Question] [answered]
So following along with [**Testing Hasura GraphQL APIs with Karate**](https://hasura.io/blog/testing-hasura-graphql-apis-with-karate/) guide, I download the [Testing Hasura API Karate](https://github.com/martin-hasura/testing-hasura-api-karate) repo. However I am unable to build the karate container due to an issue with the `karate/Dockerfile` provided in the sample repo.

> TL;DR It would be good that the recommended testing guides for Hasura be kept up-to-date, however I do recognize this as an issue with the Dockerfile and image being used likely being out-of-date. 

The `karate/Dockerfile` fails to build at:
```Dockerfile
RUN apt-get update && apt-get install -y wget && rm -rf /var/lib/apt/lists/*
```

Resulting error:
```bash
failed to solve: process "/bin/sh -c apt-get update && apt-get install -y wget && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100
```

Snippet of failed `wget`:
```bash
#0 3.078 Ign:14 http://deb.debian.org/debian stretch-updates/main arm64 Packages
#0 3.161 Ign:15 http://deb.debian.org/debian stretch-backports/main all Packages
#0 3.241 Ign:16 http://deb.debian.org/debian stretch-backports/main arm64 Packages
#0 3.319 Err:11 http://deb.debian.org/debian stretch/main arm64 Packages
#0 3.319   404  Not Found
#0 3.411 Ign:12 http://deb.debian.org/debian stretch/main all Packages
#0 3.488 Ign:13 http://deb.debian.org/debian stretch-updates/main all Packages
#0 3.563 Err:14 http://deb.debian.org/debian stretch-updates/main arm64 Packages
#0 3.563   404  Not Found
#0 3.658 Ign:15 http://deb.debian.org/debian stretch-backports/main all Packages
#0 3.739 Err:16 http://deb.debian.org/debian stretch-backports/main arm64 Packages
#0 3.739   404  Not Found
#0 3.750 Reading package lists...
#0 3.767 W: The repository 'http://security.debian.org/debian-security stretch/updates Release' does not have a Release file.
#0 3.767 W: The repository 'http://deb.debian.org/debian stretch Release' does not have a Release file.
```

**Comment by m-rgba** [comment, marked as answer]
Hey there! I'm the original author of that post.
I had originally pinned the deps and stuff in that post to try and give it some longevity, but it looks like some things broke.

Lemme have a look over the weekend and I'll try to get it up and running, pretty much need to:
- Update to openjdk:11, should fix that issue above.
- Karate is also on 1.4.0 now (vs. 0.9.6 - requires opendjk 11 as well), and they have .jars they released on GitHub now.
- Hasura has update to 2.0+ (this is on 1.3.3), I may as well bump it as well while I'm in there.

**Reply by m-rgba** [reply]
Just updated the repo here: https://github.com/martin-hasura/testing-hasura-api-karate
Should work now, let me know if you run into any issues getting it running.
